src/Npgsql/NpgsqlConnection.cs

```diff
@@ -201,7 +201,25 @@ void OpenInternal()
             // have Password populated and should not be overwritten.
             if (Password == null)
             {
-                Password = Settings.Password;
+                if (Settings.Password != null)
+                {
+                    Password = Settings.Password;
+                }
+                else if (Settings.Password == null && !IntegratedSecurity)
+                {
+                    // no password was provided. attempt to pull the password from the pgpass file.
+                    var pgPassFile = new PgPassFile(PgPassFile.GetPgPassFilePath());
+
+                    var host = Settings.Host ?? PgPassEntry.PGPASS_WILDCARD;
```

Comment:
I think we always have Host and Database (just like we always have Port), so there's no need to use null coalescing here.